### ISSUE TITLE
How does node.js shut down so silently?

### ISSUE THREAD

**Issue by jsdeveloper97** [nodejs/help#1189]
* **Node.js Version**: v9.9.0
* **OS**: win10
* **Scope**: code
* **Module**: process

**addon.cc**
```cc
#include <node.h>
using namespace v8;
void init(Local<Object> exports){ ++*(int*)0; }
NODE_MODULE(NODE_GYP_MODULE_NAME, init)
```
**main.js**
```js
require('./addon');
```

If I test this c++ code in a standalone program (remove `<node.h>` header and move the code into `main` function) then it displays message `test.exe has stopped working. Please close the program`. But, if I build it with `node-gyp` and use it as node module, then node crashes silently and no message appears.

Which trick does node.js use to prevent windows from showing annoying message about the crash? I would like to implement the same behavior in my standalone programs.

Thanks.

**Comment by thekemkid** [comment]
Is node crashing, or just exiting? You're not logging/outputting anything, so it should be "silent" in the success case.

**Comment by jsdeveloper97** [comment]
> Is node crashing, or just exiting

It segfaults, of course, so I think it is considered as crashing. But, my point was that on windows node.js doesn't display message like all other programs do on crash. It just crashes without any messages, so I am wondering how to achieve it in other, standalone, programs.

**Comment by bnoordhuis** [comment]
https://msdn.microsoft.com/en-us/library/1y71x448.aspx <- Node.js (well, libuv) calls `_CrtSetReportMode()` to disable the popups.

**Comment by jsdeveloper97** [comment]
Thanks, but... that function is available only in MSVC, and I didn't find a way to use it in gcc or other compilers. Is there an equivalent in `windows.h`?

**Comment by bnoordhuis** [comment]
It's a feature of the C runtime, not the compiler.  If you mean mingw doesn't define it, just add your own definition.

**Comment by jsdeveloper97** [comment]
After a lot of research, I found the answer. The key is to call

```c++
SetErrorMode(SEM_NOGPFAULTERRORBOX);
```

Declaration of `SetErrorMode` is from `<windows.h>`. It should be called at the beginning of the program. This is the only solution that currently works for me (and also works properly in gcc).

@ bnoordhuis, I had tried your suggestion, but I couldn't make it work. It seems that `_CrtSetReportMode` is actually a macro and what it does is not clear to me, but I assume it calls `SetErrorMode` somewhere, right? Isn't it easier to use the function instead of macro?

Anyway, thanks for helping 👍
